src/BaseElement.js

```diff
@@ -346,8 +346,21 @@ class BaseElement extends HTMLElement {
 			this.requestContext(key, value);
 		});
 		// define context provider
-		if (Object.keys(this.provideProperties()).length > 0) {
+		const providedKeys = Object.keys(this.provideProperties());
+		if (providedKeys.length > 0) {
 			this.addEventListener('request-context', this.onRequestContext);
+			// check if there are already connected elements in child dom and restart requests
+			getAllElementChildren(this).forEach((customChild) => {
```

Comment:
@quarkus 'this' is probably not right. It should be this.getRoot(), no? But what about slotted children? Are they technically children of the element they are slotted into?!

Reply:
Absolutely!

I think we had the light vs shadow DOM Discussion in some Luna Context and agreed that light dom belongs to the element and not to the shadow element that owns the slots.

Anyways .. this should probably be tested as its actually interisting which way the request event  travels ...